src/ts/edit.ts

```diff
@@ -469,4 +469,52 @@ document.addEventListener('DOMContentLoaded', () => {
     const content = (document.getElementById('date_input') as HTMLInputElement).value;
     EntityC.update(entity.id, Target.Date, content);
   });
+
+  // this should be in uploads but there is no good way so far to interact with the two editors there
+  document.getElementById('filesdiv').querySelectorAll('[data-action="replace-uploaded-file"]').forEach(el => {
+    el.addEventListener('submit', event => {
+      event.preventDefault();
+
+      // we can identify an image by the src attribute in this context
+      const searchPrefixSrc = 'src="app/download.php?f=';
+      const searchPrefixMd = '![image](app/download.php?f=';
+      const formElement = event.target as HTMLFormElement;
+      const editorCurrentContent = editor.getContent();
+
+      // submit form if longName is not found in body
+      if ((editorCurrentContent.indexOf(searchPrefixSrc + formElement.dataset.longName) === -1)
+        && (editorCurrentContent.indexOf(searchPrefixMd + formElement.dataset.longName) === -1)
+      ) {
+        formElement.submit();
+        return true;
+      }
+
+      const formData = new FormData(formElement);
+      formData.set('extraParam', 'noRedirect');
+      fetch('app/controllers/RequestHandler.php', {
```

Comment:
why not use `postForm` from `Ajax.class` here? https://github.com/elabftw/elabftw/blob/master/src/ts/Ajax.class.ts#L65

Reply:
Because currently all the relevant information is nicely provided via the hidden html input elements and this information is easily converted to a FormData object. In addition the submit() function still needs the html form information.
If we use postForm() we need to provide the payload object so all information needs to be converted. We still need the file input element in the form but need to convert that first to a File object and than add it to the payload only to convert it back to a FormData object.

Maybe there is a more elegant solution but right now I do not see it. I was thinking about to add an upload info div with all the relevant data attributes so the information does not need to be repeated several times for the different html elements. But than in JS that info needs to be made available. Not sure if that will make it easier.

Reply:
ok, makes sense.

Reply:
I wanted to avoid something like this https://github.com/elabftw/elabftw/blob/3195dfd2280c23ae90f6c3ad3559db1793810685/src/ts/editusers.ts#L35-L40